dkg/frost/frost_test.go

```diff
@@ -0,0 +1,181 @@
+package frost
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/bloxapp/ssv-spec/dkg"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+)
+
+func TestFrost2_4(t *testing.T) {
+	requestID := dkg.RequestID{}
+	for i := range requestID {
+		requestID[i] = 1
+	}
+
+	operators := []types.OperatorID{
+		1, 2, 3, 4,
+	}
+
+	mockNetwork := MockNetwork{
+		nodes: make(map[types.OperatorID]*Node),
+	}
+
+	dkgsigner := testingutils.NewTestingKeyManager()
+	storage := testingutils.NewTestingStorage()
+
+	for _, operator := range operators {
+		operatorID := types.OperatorID(operator)
+
+		node := newNode(operatorID)
+		mockNetwork.nodes[operatorID] = node
+	}
+
+	kgps := make(map[uint32]dkg.KeyGenProtocol)
+
+	for _, operatorID := range operators {
+		p := New(&mockNetwork, operatorID, requestID, dkgsigner, storage)
+		kgps[uint32(operatorID)] = p
+
+		mockNetwork.nodes[operatorID].SetProcessMsgFn(p.ProcessMsg)
+	}
+
+	for _, node := range mockNetwork.nodes {
+		go func() {
+			_ = node.Run()
+		}()
+		defer node.Exit()
+	}
+
+	threshold := 2
+
+	g := errgroup.Group{}
```

Comment:
we don't use async in testing, just have a bunch of msgs go to the frost implementation. Makes it easier to read